DifferentiationInterface/ext/DifferentiationInterfaceReactantExt/onearg.jl

```diff
@@ -0,0 +1,81 @@
+struct ReactantGradientPrep{SIG, XR, GR, CG, CG!, CVG, CVG!} <: DI.GradientPrep{SIG}
+    _sig::Val{SIG}
+    xr::XR
+    gr::GR
+    compiled_gradient::CG
+    compiled_gradient!::CG!
+    compiled_value_and_gradient::CVG
+    compiled_value_and_gradient!::CVG!
+end
+
+function DI.prepare_gradient_nokwarg(strict::Val, f::F, rebackend::AutoReactant, x) where {F}
+    _sig = DI.signature(f, rebackend, x; strict)
+    backend = rebackend.mode
+    xr = to_rarray(x)
```

Comment:
you shoudl check if x / gr are already reactant arrays, if so we should not call to_rarray and/or copy

Reply:
I was wondering whether the copy back from RArray was necessary at all in the general case. Are there limitations on what Reactant arrays can do?

Reply:
But if that's not yet the case, can't we add that one line to Reactant? Is there an issue here that I don't see?

Reply:
The semantics of to_rarray currently does not guarantee if the object returned will be the same or a copy, and we shouldn't rely on it since Reactant may choose to specify this differently in a non breaking change